tests/ga/test_policy_engine.py

```diff
@@ -15,20 +15,516 @@
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {}
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                            msg="Global signatureRequired policy is false, so signature should not be enforced.")
+
+    def test_should_enforce_signature_if_global_signatureRequired_true_and_individual_signatureRequired_not_specified(self):
+        """
+        If individual policy is present, but signatureRequired is not specified for that policy, use global signatureRequired.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": True,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertTrue(should_enforce_signature,
+                            msg="Individual signatureRequired policy is not set, so should use global policy and enforce signature.")
+
+    def test_should_not_enforce_signature_if_global_signatureRequired_false_and_individual_signatureRequired_not_specified(self):
+        """
+        If individual policy is present, but signatureRequired is not specified for that policy, use global signatureRequired.
+        """
+        test_extension = Extension(name=TEST_EXTENSION_NAME)
+        policy = \
+            {
+                "policyVersion": "0.1.0",
+                "extensionPolicies": {
+                    "allowListedExtensionsOnly": True,
+                    "signatureRequired": False,
+                    "extensions": {
+                        TEST_EXTENSION_NAME: {}
+                    }
+                }
+            }
+        self._create_policy_file(policy)
+        engine = ExtensionPolicyEngine()
+        should_enforce_signature = engine.should_enforce_signature_validation(test_extension)
+        self.assertFalse(should_enforce_signature,
+                            msg="Individual signatureRequired policy is not set, so should use global policy and enforce signature.")
+    def test_extension_name_in_policy_should_be_case_insensitive(self):
```

Comment:
nit: there should be a new line after this test

Reply:
Fixed!